### DISCUSSION TITLE
Check all rows from header

### DISCUSSION THREAD

**Discussion by danisal** [nick-keller/react-datasheet-grid#186] [Q&A] [answered]
Hi, is it possible to access the `rowData` and `setRowData` on the column header cell? I'm using a custom title component, `CustomTitleComponent` but I don't have the props out-of-the-box. Is possible to pass those to the component?

**Comment by nick-keller** [comment, marked as answer]
Hello @danisal ,
`rowData` is usually the data of the row, but in the case of the header it has no row. But you could pass the entire data to your header component (either by props, or using a context) to you'd be able to display things based on that.

**Reply by danisal** [reply]
My use case is to have a checkbox on a column header, in order to (un)check all row's checkboxes. Any tips/suggestions?

![image](https://user-images.githubusercontent.com/9869755/190372862-bce58874-cca9-4c9a-a774-c2af60c368ee.png)

**Reply by nick-keller** [reply]
It should be pretty straight forward:
```tsx
const [data, setData] = useState([])
const columns = [
  { title: <CheckBox onClick={() => setData(data.map(/* set value to true for each row */))} />},
  // ...
]

return <DataSheetGrid value={data} columns={columns} />
```
